### ISSUE TITLE
Weapon Skill not Blizzlike [Very Important]

### ISSUE THREAD

**Issue by Hanzemon** [LightsHope/issues#1096]
**Introduction.**

This will be a long article about the flaws in the current attack table and the mechanics involved, I will briefly touch the current behavior and compare it to the Blizzlike Classic behavior by going deeply into the mechanics involved, Any error or lack of information regarding the current behavior should not be taken too strongly as I will provide the correct behavior that will kick the other into the maelstrom forever.

--------------------------------------------**THE CURRENT BEHAVIOR**-------------------------------------------

Right now the effects of Weapon skill over the Miss chance **against a mob 3 levels higher**  looks like this:

Weapon skill: 300  --->    Chance to miss: 8.60% 
Weapon skill: 301  --->    Chance to miss: 7.96%
Weapon skill: 302  --->    Chance to miss: 7.32% 
Weapon skill: 303  --->    Chance to miss: 6.66% --Hail satan 
Weapon skill: 304  --->    Chance to miss: 6.04% 
Weapon skill: 305  --->    Chance to miss: 5.40%   

_Every 1 weapon skill is reducing 0.64% chance to miss the target, this only occurs when the difference between the mob's defense skill and player's weapon skill is > 10._

Now beyond 305 it changes and the effects of weapon skill over the miss chance against a mob 3 levels higher looks like this:

Weapon skill: 306  --->    Chance to miss: 5.36%
Weapon skill: 307  --->    Chance to miss: 5.32%
Weapon skill: 308  --->    Chance to miss: 5.28% 
Weapon skill: 309  --->    Chance to miss: 5.24%
Weapon skill: 310  --->    Chance to miss: 5.20%
Weapon skill: 311  --->    Chance to miss: 5.16% 
Weapon skill: 312  --->    Chance to miss: 5.12%
Weapon skill: 313 --->     Chance to miss: 5.08% 
Weapon skill: 314  --->    Chance to miss: 5.04% 
Weapon skill: 315  --->    Chance to miss: 5.00% 

_Now every 1 weapon skill reduces the chance to miss the target by 0.04%._

[Table](https://media.discordapp.net/attachments/271013458139414528/344830038807412737/unknown.png?width=766&height=667)

---------------------------------------**INTENDED BEHAVIOR (BLIZZLIKE)**---------------------------------------

**In the original version of World of Warcraft it worked like this:**

--Citing Wow Gamepedia on _Weapon skill_ September 30th 2006.

_For each point that your weapon skill exceeds your opponent's defense, you gain the following:_

_Your chance to miss decreases by 0.04%._
_Your chance to score a critical hit increases by 0.04%._
_Your opponent's chance to block your attack decreases by 0.04%._
_Your opponent's chance to parry your attack decreases by 0.04%._
_Your opponent's chance to dodge your attack decreases by 0.04%._

_Effectively, you can consider weapon skill to be the inverse of the defense skill_. 

Source: [Wow Gamepedia on Weapon Skill  30 September, 2006.](https://wow.gamepedia.com/index.php?title=Weapon_skill&oldid=267725)

-

 **Now, that brings some doubts, it specifies "exceeds". But what about when it's lower?. 
Let's have some more insight:**

--Citing Wow Gamepedia on _Dual wield_ September 6th 2006.

...Take a rogue for example vs. a 63 monster, he has 5%+(0.04% * 15)= 5.6% base miss chance (assuming he has 300 weapon skill and the monster has no special internal Defense skill bonuses),** plus 19% dual wield miss chance, leaving us with a total miss chance of 24.6%.

**We can see it has actually the same effect**

Source: [Wow Gamepedia on Dual wield 6 September, 2006.](https://wow.gamepedia.com/index.php?title=Dual_wield&oldid=236329)

-

**Okay, now we know that the miss chance against a lvl 63 mob at 300 weapon skill is 5.6%  and not 8.6%, but, what does the actual **Attack Table** look like?**

**Let's jump into that:**

--Citing WoWWiki on Attack table October 29th 2006.

_Example 2
Elliot the Rogue is level 60, and is attacking a boss mob along with the rest of his raid group. He has stocked up on all sorts of +crit gear, so his crit chance against the boss mob is a whopping 30%. Since he is dual-wielding, he incurs an additional +19% miss chance on every autoattack swing; and although his +5 weapon skill gear does counter a little bit of this, he did not bring any +hit gear with him, so his miss chance against a boss mob is 24.4%. (+Hit gear doesn't directly increase your chance of a hit; instead, it reduces your chance of a miss.) When he attacks the boss mob from the front, Elliot's attack table looks like this:_

![example 2](https://user-images.githubusercontent.com/32701428/48597024-fc474a80-e931-11e8-8bf1-35bbfc83eda0.PNG)

_Note that despite his nominal "crit chance" of 30%, his miss chance from dual-wielding and his glancing blow chance due to the boss's high level mean that only 19.4% of his attacks are going to be critical hits. This conundrum is sometimes referred to as the Crit Cap -- his chance to crit is capped at 19.4%, so 10.6% of his +crit bonus is wasted._

**Hanzemon: Now we can notice that Elliot's chance to miss is 24.40% instead of 24.60% (5% base + 0.6% Wepskill/Defskill difference + 19% dual wield) this is because as said before he has +5 Weapon skill which reduces his chances to miss by: 5 * 0.04% = 0.20%; now substract it to the total miss chance: 24.60% - 0.20% = 24.40%.** **Let's continue a lilttle further.** 

_Realizing his stupidity, Elliot maneuvers around behind the boss mob to continue his attacks. From behind, a mob cannot parry or block, so Elliot's attack table now looks like this:_

![example 2 1](https://user-images.githubusercontent.com/32701428/48597053-26007180-e932-11e8-8b4c-e2382a8a06d6.PNG)

**And finally but not less important, for yellow attacks**

![image](https://user-images.githubusercontent.com/32701428/48596861-306e3b80-e931-11e8-8b98-63f1be1ff8be.png)

Source: [WoWWiki on Attack table October 29th 2006](http://wowwiki.wikia.com/wiki/Attack_table?oldid=303353)

------------------------------------------------------**EVIDENCE**-----------------------------------------------------

**Now I will show you the post of a player that played World of Warcraft back in 2006.**

--Citing the player Zoroaster member of < The Obsidian Order > from Aggramar server in the forums of WoW Europe.

 **Regarding Avoidance:**

_Dodge, Block and Parry all have a base stat of 5% for mobs. They increase as the mobs Defense becomes higher than your weapon skill at 0.04 per point. Each Mob gains 5 points in defence per level. Block and Parry only apply when attacking from the front. _

_Base miss chance is 5%. If you duel weild your white attacks (auto attacks) suffer a further 19%. All your +hit gear reduces this number. For every point the mob defense is higher than your weapon skill, miss goes up by 0.04. _

**His examples of the Attack table**

_Your yellow strikes (special attacks) aren't that effected by targetting the boss, but your white strikes (auto attacks) all of a sudden are much less damaging. Worse to come - say you have +5% Hit and 25% crit, fresh from farming level 60 mobs..._ 

Yellow (Special) 
5.6% Dodge 
5.6% Parry (0) 
5.6% Block (0) 
0.6% Miss 
24.4% Crit (-0.6 due to weapon/defence differnece) 
58.2% Hit (69.4) 

White (Auto) 
5.6% Dodge 
5.6% Parry (0) 
5.6% Block) (0) 
19.6% Miss 
40% Glancing 
23.6% Crit (24.4) 
0% Hit (10.4) 

Source & complete post (I recommend you to read it): http://web.archive.org/web/20061115223930/http://forums.wow-europe.com/thread.html?topicId=14381707&sid=1

---------------------------------------------GLANCING BLOW DAMAGE----------------------------------------

**To finish let's go over glancing blows real quick**

--Citing Wow Gamepedia on _Weapon skill_ September 30th 2006.

_Glancing Blow chance
The percentage chance of a glancing blow on any melee auto-attack, with either the main hand weapon or the off-hand weapon, is normally agreed to be the following:_

_(NPC’s Defense skill – YOUR Weapon skill – 5) * 3% + 10%_

**_... where "YOUR Weapon skill" is capped at your level * 5._**

**With this formula we can calculate the glancing blow chance very easily, which is 40% against bosses regardless of all the +Weapon Skill bonuses you might have, including racials.**

Source: [Wow Gamepedia on Weapon Skill  30 September, 2006.](https://wow.gamepedia.com/index.php?title=Weapon_skill&oldid=267725).

**Thanks to Killerduki, for debating me on the behavior of the current private servers in comparison with all the data and evidence from retail Classic World of Warcraft** .

-Hanzemon.

**Comment by Hanzemon** [comment]
Also.

![captura](https://user-images.githubusercontent.com/32701428/48666847-0f782880-eaa0-11e8-8058-21d338c21256.PNG)

Source: [Wow Gamepedia on Weapon Skill 30 September, 2006.](https://wow.gamepedia.com/index.php?title=Weapon_skill&oldid=267725)

And also.

![captura2](https://user-images.githubusercontent.com/32701428/48666887-ba88e200-eaa0-11e8-936d-e8b652fd009d.PNG)

Source: [Wow Gamepedia on Glancing Blows 23 September 2006.](https://wow.gamepedia.com/index.php?title=Glancing_blow&oldid=382350)

**Comment by Comfarol** [comment]
Wrong.

**Comment by Holyfrogg** [comment]
A wiki is a poor source imo. Not only can be it edited easily, but it's also written by people who may or may not have a good understanding of the underlying mechanics.

I also get some signal flares going off when I see Killerduki being behind this. I've seen that guy on reddit and the wow forums for years and years arguing that Paladins are the best class in the game and that warriors are only better on private servers because of buggy mechanics like weapon skills being implemented correctly, 

I dont mean to use this bugtracker to bash anyone, but I'm simply trying to illustrate a point.

https://eu.battle.net/forums/en/wow/topic/17624042278?page=2#post-36
Here's Duki using a screenshot of himself, and a video of a paladin wearing sulfuras topping DPS as proof that Paladins are a top DPS class. If this is indicative of the research methodology he is using for collecting evidence then I would be very skeptical of his conclusions.

He's clearly biased, and is personally invested in seeing other physical DPS be nerfed.

**Comment by Hanzemon** [comment]
Wow wikis are known for documenting the data from the game in its time by capturing and testing every detail, I provide different instances of information from different sources that are not correlated with eachother. As an example compare the evidence from "**[Wow Gamepedia on Dual wield 6 September, 2006.](https://wow.gamepedia.com/index.php?title=Dual_wield&oldid=236329)**" and "**[WoWWiki on Attack table October 29th 2006](http://wowwiki.wikia.com/wiki/Attack_table?oldid=303353)**"; those are 2 different wikis documentating 2 different aspects of the game since its earliest days, if you check the history of those threads you can go to the earlier ones and see when the information was pending for test, you can see the results of the tests individually and observe the evolution of those posts individually based on evidence from back in the day. The results are similar all across the board, you can even check the [thread a player made in EU Forums in 2006](http://web.archive.org/web/20061115223930/http://forums.wow-europe.com/thread.html?topicId=14381707&sid=1), he himself tests the resuts of attacking mobs and gets to the same conclusions as both of the wikis.

All these 3 sources of evidence match perfectly with the calculations provided by  [Wow Gamepedia on Weapon Skill  30 September, 2006.](https://wow.gamepedia.com/index.php?title=Weapon_skill&oldid=267725). In which is claimed that: 

**For each point that your weapon skill exceeds your opponent's defense, you gain the following:**

**Your chance to miss decreases by 0.04%.**
**Your chance to score a critical hit increases by 0.04%.**
**Your opponent's chance to block your attack decreases by 0.04%.**
**Your opponent's chance to parry your attack decreases by 0.04%.**
**Your opponent's chance to dodge your attack decreases by 0.04%.**

**Effectively, you can consider weapon skill to be the inverse of the defense skill.**

Which matches perfectly with player's tests from 2005 and 2006 reflected in the evidence provided.

The data and evidence provided in this post are empirical, wether you disagree with someone who might be partially involved in this post doesn't change the data provided or its validity.

**Comment by Hanzemon** [comment]
@Holyfrogg Since it is clear that you have not read any of the data provided in this thread, anyone who can read and understand what's said in this post can get to the conclusion that iti s based on empirical evidence. Yes you are using this thread specifically to bash someone.

**Comment by Holyfrogg** [comment]
I'm sorry, but all I see are forum posts that wikis that are well known to be inaccurate due to the nature of a wiki.

It may very well be that the data is correct, but considering the person who helped put it together is well known for cherry picking whatever data will buff his class the most, and have done things like spend years trying to change Seal of Righteousness coefficients based on a wiki from TBC, it is very relevant to mention this and does not make it bashing. Just making sure whoever is reading this bug report is informed of all the facts.

At the end of the day, the only source here is just a wiki written by some player somewhere and a forum post with similar data, and unless there's a link in the wiki/post to show how the data was obtained, it really is just a big wall of text that may or may not be correct.

**Comment by Hanzemon** [comment]
**_WIKIS ARE NOT KNOW FOR BEING INNACURATE AT ALL, ALL THE CONTRARY. IT IS THE MAIN SOURCE OF DATA FOR WORLD OF WARCRAFT CREATED FOR PLAYERS BY PLAYERS BASED ON EMPIRICAL DATA, EVIDENCE AND LOTS OF TESTS OVER THE YEARS._**

 Again, it is clear that you have either NOT read this post, nor the sources of information, nor the evidence provided no only from wikis but from players tests. Or you are just a troll. And again, you are using this to discredit someone who you disagree with. You have absolutely no proof or data to discredit anyone or any wiki, you haven't researched anything and don't know anything about this topic AT ALL.

This post was made by me and only me, it took me hours to gather all the sources of information I could find and then redact the post and make it clear for the reader, the only contribution that didn't came directly from me was the EU wow forum post and a discussion on the behavior of the current private servers vs blizzard original's.

**Comment by robmonte** [comment]
Empirical evidence is still evidence. Many people seem to disregard data far too easily because they like to throw around terms like "confirmation bias" and "anecdotal evidence" without fully understanding what they imply.

In this case, just because data is based on observation and not developer-provided fact doesn't mean it isn't worthwhile information. You can't discredit something just because "I don't like this source." You need evidence to disprove this data. Is there conflicting data and proof about how weapon skill and glancing blows work? 

In other words, how was Lights Hope/Elysium/Nostalrius decided to handle these mechanics, and what is it now? Was it the devs just using their best judgement? Was it community feedback on forums? Either one is far *more* anecdotal than this wiki page is.

**Comment by Hanzemon** [comment]
@robmonte I assume it was the developers desicion, they missinterpreted the information they were provided regarding weapon skill and assumed 5 weapon skill gave you 3%. Because they compared the attack table from 2009 and the attack table from 2006. 

They assumed this [statement made on 2006 in the wowwiki/wowgamepedia Wikis](http://wowwiki.wikia.com/wiki/Attack_table?oldid=303353) :  Elliot the Rogue is level 60, and is attacking a boss mob along with the rest of his raid group. He has stocked up on all sorts of +crit gear, so his crit chance against the boss mob is a whopping 30%. Since he is dual-wielding, he incurs an additional +19% miss chance on every autoattack swing; **and although his +5 weapon skill gear does counter a little bit of this**...

Resulting on elliot having 24.40% chance to miss as you can see here:
![](https://user-images.githubusercontent.com/32701428/48597053-26007180-e932-11e8-8b4c-e2382a8a06d6.PNG)
Meant (for them) that 5+ Weapon skill gave 3% hit chance. (24.40% instead of 24.60%, which they mistakenly assumed was instead of 27%) 

Since in the [2009s-2010s Wowwiki's/Gamepedias wikis](http://wowwiki.wikia.com/wiki/Attack_table?oldid=1962391) it is shown that  the miss chance is 27% and the +weapon skill part has been removed.
![captura](https://user-images.githubusercontent.com/32701428/48965080-be2ed400-ef8c-11e8-9a6f-2fcc2d3049ea.PNG)

That would explain [This](https://media.discordapp.net/attachments/271013458139414528/344830038807412737/unknown.png?width=766&height=667)

**Comment by Ethugd** [comment]
http://wowwiki.wikia.com/wiki/Talk:Weapon_skill I believe this is the main source they used

> -- 13:47, 3 August 2006 (EDT)2 things that aren't accurate in the Wiki article:
> 
> "A glancing blow only occurs on white damage, can not crit and has a damage reduction proportional to the difference between the attacker's weapon skill and the target's defense."
> 
> That's not confirmed. All we know is that we have 40% chance of getting a glancing blow on non-special attacks; and that the damage reduction becomes ~0% when you reach 310 weapon skill. If you say proportional, that would mean you're saying raid bosses (who we all suppose are level 63) have 310 defense. That is NOT the case.
> 
> 
> Second:
> 
> "With no +skill/+hit gear on, a dual wielder vs a level 63 mob will have the following damage table from the front: 5.6% dodge, 5.6% parry, 24.6% miss, 5.6% block, 40% glancing blow, 18.6% combined hit + crit. Attacking from the back prevents block and parry giving 29.8% combined hit + crit. Attacks from the back can still be dodged."
> 
> Is flat out wrong. The numbers come from thinking "since it's a raidboss lvl 63, he has 315 defense". If that was all he had, and no additional modifiers, +6 toHit in form of talents and/or gear would be enough to never miss a special attack. It is not.
> 
> 315 defense theory wrong
> 
> That is a screenshot with 5/5 precision, and +1 to hit gear vs. Onyxia lvl 63. Note the 4 Sinister Strike misses in recap. Also note that recap does not count dodges or parries from onyxia as misses - the /combatlog from that fight confirms the 4 misses. That proves that 315 defense - 5,6% miss chance is not all there is against raidbosses.
> 
> Collected numbers seem to point at +8 to Hit with gear or talents to be enough to never miss a special against a boss. Numbers i collected show no misses in over 20 million points of damage over several weeks. While that doesn't prove +8 is all you need (it could be a statistical fluke making me not miss, although there is a 0,6% misschance (random number, only for the example), what is proven is that raidbosses do not have 315 defense only with no additional things being factored in.
> 
> That leaves at least 2 choices: 1) raidbosses have 300 defense, and cause an artificial 3% misschance. (having 315 defense +3% would very likely have shown a miss until now) 2) Raidbosses come with a bonus 60 defense, raising it to 375 - which leads to the observed needed +8 toHit to never miss, and is in a way supported by the high parry and dodge chances bosses seem to have (although that is as of now an impression, and not backed up by numbers)
> 
> Unless there is feedback or comments soon, I'll implement my findings into the Wiki page, because right now it's spreading disinformation. Also: "nice findings" is cute and all, but I'd prefer a proof that i am wrong if i am.
> 
> Small note: this information does not invalidate the findings about crit/hit vs glancing blows.
> 
> --- Assassinette

**Comment by Hanzemon** [comment]
All it sugests by now is that either some raids bosses have special internal Defense Skill Bonuses or that the player wasn't capped at 300 at the moment of testing.

That case, however, is taken into account in the Dual Wield post:

"Citing Wow Gamepedia on Dual wield September 6th 2006.

...Take a rogue for example vs. a 63 monster, he has 5%+(0.04% * 15)= 5.6% base miss chance **(assuming he has 300 weapon skill and the monster __has no special internal Defense skill bonuses__)**."

Since it is unlikely that a rogue raiding Onyxia is not already capped to 300, it suggests that some bosses indeed had some +Defense Skill Bonuses, just as players with +Defense gear. As this case might most likely be possible we should still stick to the original formulas since those are proven to be  systematically accourate out of the domain of the exceptions.

This is interesting but in my personal opinion I think it belongs more to bosses stat values rather than values regarding the most fundamental mechanics which were indeed figured out throught tests as well.

Modifying the formulas based on a single tests from a single boss would be like changing the programming language for an issue on a single line.
